galley/pkg/kube/converter/converter_test.go

```diff
@@ -606,3 +608,89 @@ func TestShouldProcessIngress(t *testing.T) {
 		}
 	}
 }
+
+func TestKubeServiceResource(t *testing.T) {
+	cases := []struct {
+		name string
+		from corev1.Service
+		want proto.Message
+	}{
+		{
+			name: "Simple",
+			from: corev1.Service{
+				ObjectMeta: meta_v1.ObjectMeta{
+					Name:              "reviews",
+					Namespace:         "default",
+					CreationTimestamp: meta_v1.Time{Time: fakeCreateTime},
+				},
+				Spec: corev1.ServiceSpec{
+					ClusterIP: "10.39.241.161",
+					Ports: []corev1.ServicePort{
+						{
+							Name:       "http",
+							Protocol:   "TCP",
+							Port:       9080,
+							TargetPort: intstr.FromInt(9080),
+						},
+						{
+							Name:       "http-web",
+							Protocol:   "TCP",
+							Port:       9081,
+							TargetPort: intstr.FromInt(9081),
+						},
+						{
+							Name:       "ssh",
+							Protocol:   "TCP",
+							Port:       9082,
+							TargetPort: intstr.FromInt(9082),
+						},
+					},
+				},
+			},
+			want: &networking.ServiceEntry{
+				Hosts:     []string{"reviews.default.svc.cluster.local"},
+				Addresses: []string{"10.39.241.161"},
+				Ports: []*networking.Port{
+					{
+						Name:     "http",
+						Number:   9080,
+						Protocol: "HTTP",
+					},
+					{
+						Name:     "http-web",
+						Number:   9081,
+						Protocol: "HTTP",
+					},
+					{
+						Name:     "ssh",
+						Number:   9082,
+						Protocol: "TCP",
+					},
+				},
+			},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			var u unstructured.Unstructured
+			u.Object = make(map[string]interface{})
+			if err := convertJSON(&c.from, &u.Object); err != nil {
+				t.Fatalf("Internal test error: %v", err)
+			}
+			want := []Entry{{
+				Key:          resource.FullNameFromNamespaceAndName(c.from.Namespace, c.from.Name),
+				CreationTime: fakeCreateTime.Local(),
+				Resource:     c.want,
+			}}
+			got, err := kubeServiceResource(nil, resource.Info{}, want[0].Key, "kind", &u)
+			if err != nil {
+				t.Fatalf("kubeServiceResource: %v", err)
+			}
+			// TODO: Use cmp.Equal+cmp.Diff?
```

Comment:
reflect.DeepEqual generally works fine, and it is one less dependency.

Reply:
https://godoc.org/github.com/google/go-cmp/cmp#Diff may be worthwhile for pretty printing the diffs. There are probably other places in the code where the filtering options would be useful.

Reply:
We already have spew.Sdump. We should pick one and align on it preferrably.

Reply:
It is one more additional dependency. We already have an abundance of dependencies at this point. I'd rather avoid introducing new dependencies unless it is used across the whole source tree.